src/deepqmc/wf/paulinet/omni.py

```diff
@@ -1,158 +1,161 @@
-            **(schnet_kwargs or {}),
-        )
-        if with_jastrow:
-            self.jastrow = get_log_dnn(embedding_dim, 1, SSP, n_layers=n_jastrow_layers)
-        else:
-            self.forward_jastrow = None
-        if with_backflow:
-            backflow = [
-                get_log_dnn(
-                    embedding_dim,
-                    n_orbitals,
-                    SSP,
-                    n_layers=n_backflow_layers,
-                    last_bias=True,
+        self.dnn_core = nn.ModuleDict(
+            {
+                'many-body': many_body_dnn_factory(
+                    n_up,
+                    n_down,
+                    len(mol.coords),
+                    dist_feat_dim=dist_feat_dim,
+                    **many_body_dnn_kwargs,
                 )
-                for _ in range(n_channels)
-            ]
-            self.backflow = nn.ModuleList(backflow)
-        else:
-            self.forward_backflow = None
-        if with_r_backflow:
-            self.r_backflow = Backflow(mol, embedding_dim)
-        else:
-            self.forward_r_backflow = None
-        self._cache = {}
-
-    def _get_embeddings(self, edges_elec, edges_nuc):
-        edges_id = id(edges_elec) + id(edges_nuc)
-        if self._cache.get('dist_feats_id') != edges_id:
-            self._cache['dist_feats_id'] = edges_id
-            self._cache['embeddings'] = self.schnet(edges_elec, edges_nuc)
-        return self._cache['embeddings']
-
-    def forward_jastrow(self, edges_elec, edges_nuc):
-        """Evaluate Jastrow factor."""
-        xs = self._get_embeddings(edges_elec, edges_nuc)
-        J = self.jastrow(xs.sum(dim=-2)).squeeze(dim=-1)
-        return J
-
-    def forward_backflow(self, edges_elec, edges_nuc):
-        """Evaluate backflow."""
-        xs = self._get_embeddings(edges_elec, edges_nuc)
-        xs = torch.stack([bf(xs) for bf in self.backflow], dim=1)
-        return xs
-
-    def forward_r_backflow(self, rs, edges_elec, edges_nuc):
-        xs = self._get_embeddings(edges_elec, edges_nuc)
-        return self.r_backflow(rs, xs)
-
-    def forward_close(self):
-        """Clear the cached SchNet embeddings."""
-        self._cache.clear()
+                if many_body_dnn_factory
+                else None,
+                'mean-field': mean_field_dnn_factory(
+                    n_up,
+                    n_down,
+                    len(mol.coords),
+                    dist_feat_dim=dist_feat_dim,
+                    **mean_field_dnn_kwargs,
+                )
+                if mean_field_dnn_factory
+                else None,
+            }
+        )
+        self.dist_basis = (
+            DistanceBasis(dist_feat_dim, cutoff=dist_feat_cutoff, envelope='nocusp')
+            if 'edges' in jastrow_stream + backflow_stream
+            or 'emb' in jastrow_stream + backflow_stream
+            else None
+        )
+        self.streams = {
```

Comment:
the `streams` dict is only used in the initializer, so it's not needed as an attribute?